lib/rdkafka/consumer.rb

```diff
@@ -60,6 +60,42 @@ def unsubscribe
       end
     end
 
+    # Pause producing or consumption for the provided list of partitions
+    #
+    # @param list [TopicPartitionList] The topic with partitions to pause
+    #
+    # @raise [RdkafkaError] When pausing subscription fails.
+    #
+    # @return [nil]
+    def pause(list)
+      unless list.is_a?(TopicPartitionList)
+        raise TypeError.new("list has to be a TopicPartitionList")
+      end
+      tpl = list.to_native_tpl
+      response = Rdkafka::Bindings.rd_kafka_pause_partitions(@native_kafka, tpl)
+      if response != 0
+        raise Rdkafka::RdkafkaError.new(response, "Error pausing '#{list.to_h}'")
+      end
+    end
+
+    # Resume producing consumption for the provided list of partitions
+    #
+    # @param list [TopicPartitionList] The topic with partitions to pause
+    #
+    # @raise [RdkafkaError] When resume subscription fails.
+    #
+    # @return [nil]
+    def resume(list)
+      unless list.is_a?(TopicPartitionList)
+        raise TypeError.new("list has to be a TopicPartitionList")
+      end
+      tpl = list.to_native_tpl
+      response = Rdkafka::Bindings.rd_kafka_resume_partitions(@native_kafka, tpl)
+      if response != 0
+        raise Rdkafka::RdkafkaError.new(response, "Error resume '#{list.to_h}'")
```

Comment:
I'd like to add an optional list field to the error class, so you can put the actual object in there for further inspection.

Reply:
Or maybe subclass error to `RdkafkaErrorWithTpl` or something like that.

Reply:
I've carefully looked at `TopicPartitionList` and I think  the first we need is a flattened  representation of `TopicPartitionList` which would be the array of TopicPartition structures

```
class TopicPartition
  attr :topic
  attr :partition
  optional :offset
  optional :err_t
end
```

So, per a partition error could be represented as `Hash<TopicPartition, RdkafkaError>`, which could be used inside `RdkafkaErrorWithTpl `

Reply:
That would indeed be nice, I'm not a huge fan of how rdkafka combines so many pieces of information in a single struct. Would make sense to create a dedicated object.